### PR TITLE
Update Hassio with security options

### LINKED ISSUE #1710: RFC: Hass.io UI
## Add-on install
- Show security rating as badge
- Show badge for network/full_access/homeassistant_api/hassio_api/docker_api/host_pid
- Show install button only on supported platforms. (use `available` flag)

## Add-on page
- Show security rating as badge
- Show red warning banner if the protection mode is disabled
- Allow to disable the protection mode
- Add support for stats to see the current resource

## Add-on store
- Gray out Add-on they not match arch/machine (use `available` flag)

## Snapshot
- Add support for upload (need change on component)

### PR DESCRIPTION
**Pull request by bramkragten**
Fixes https://github.com/home-assistant/home-assistant-polymer/issues/1710 (except the upload snapshot)

image
image

image
image
image
Unavailble addon:
image
image

### CODE REVIEW

File: hassio/src/addon-view/hassio-addon-info.js
Comment (frenck) [inline hassio/src/addon-view/hassio-addon-info.js]:
```suggestion
              <paper-button on-click="protectionToggled">Disable protected mode</paper-button>
```

**Comment by frenck** [comment]
🎉  I'm really thankful that you are doing this @bramkragten! 👍 

Added a suggestion above in a comment. Since a user actually disables the protection mode (which is on by default).

Some other small remarks:
- Maybe move down the red block right under the add-on header block.
- I like the badges, although, the avarage user might have no clue on what they mean. Maybe add some descriptions on hover? (Not sure what is possible in our UI).

File: hassio/src/addon-view/hassio-addon-info.js
Comment (frenck) [reply hassio/src/addon-view/hassio-addon-info.js]: Nevermind, I'm stupid...

**Comment by cogneato** [comment]
This looks great! Does the security rating go with the addon or does it change depending on whether the protection mode is enabled/disabled?

**Comment by frenck** [comment]
Current:
_"Protection mode is disabled, this addon can do things that can damage your system. Only use this if you know and trust the source of this addon."_

Suggestion:
_"Protection mode on this addon is disabled! This gives the add-on full access to the entire system, which adds security risks, and could damage your system when used incorrectly. Only disable the protection mode if you know, need AND trust the source of this addon."_

I think the latter explains it a little more and adds the security risk to it as well.

File: hassio/src/addon-view/hassio-addon-info.js
Comment (frenck) [inline hassio/src/addon-view/hassio-addon-info.js]: Badge description:

### Host Processes Namespace

> Usually, the processes the addon runs, are isolated from all other system processes. The addon author has requested the addon to have access to the system processes running on the host system instance, and allow the addon to spawn processes on the host system as well. This mode gives the addon full access and control to your entire Hass.io system, which adds security risks, and could damage your system when misused. Therefore, this feature impacts the addon security score negatively.

> This level of access is not granted automatically and needs to be confirmed by you. To do this, you need to disable the protection mode on the addon manually. Only disable the protection mode if you know, need AND trust the source of this addon.

**Comment by frenck** [comment]
> This looks great! Does the security rating go with the addon or does it change depending on whether the protection mode is enabled/disabled?

@cogneato That is not determined by the frontend, but by Hass.io. See: <https://github.com/home-assistant/hassio/blob/master/hassio/addons/utils.py#L17>

**Comment by frenck** [comment]
@bramkragten Missing in this list (IMHO):

- `apparmor`. This option also affects the security rating.
- `auth_api`. When enabled, the add-on is allowed to authenticate users against Home Assistant (for allowing login with your Home Assistant accounts on add-ons).

**Comment by frenck** [comment]
### AppArmor

> AppArmor ("Application Armor") is a Linux kernel security module that restricts addons capabilities like network access, raw socket access, and permission to read, write, or execute specific files.

> Addon authors can provide their security profiles, optimized for the addon, or request it to be disabled. If AppArmor is disabled, it will raise security risks and therefore, has a negative impact on the security score of the addon.

**Comment by frenck** [comment]
### Home Assistant Authentication

> An addon can authenticate users against Home Assistant, allowing add-ons to give users the possibility to log into applications running inside add-ons, using their Home Assistant username/password. This badge indicates if the add-on author requests this capability.

**Comment by tjorim** [comment]
This is already looking very nice.

I agree with Frenck to move the warning banner down, between the header and config blocks.
Furthermore, I don't think hover is the best way to show such explanation about the badges, could we use a modal?

One last thing that I think is important when speaking about the whole security aspect of add-ons: which folders can they access (config, share, ssl...) and is it read-only or R/W. 

Also, could you provide a screenshot on how non-available add-ons are shown in the store?

**Comment by pvizeli** [comment]
I love it! Thanks @frenck for the comments 👍

**Comment by cogneato** [comment]
Add-ons that are not installed?

image


image

**Comment by balloob** [comment]
lol we need a designer so badly 😉 It feels like a mismash of data thrown into a single card. I won't block this PR on it but we should think about it after this is merged. Maybe split in 2 cards: info + config/control.

Inspiration for a card based UI with a lot of info is the Play Store: https://play.google.com/store/apps/details?id=com.google.android.gm

File: hassio/src/addon-view/hassio-addon-info.js
Comment (balloob) [reply hassio/src/addon-view/hassio-addon-info.js]: I wonder if all badges should link to a help article with all the extended info

**Review by balloob — APPROVED** [review]
Code looks fine.

**Pull request merged**
